Inseting data and finding answers to business questions.txt

```diff
@@ -0,0 +1,159 @@
+ bill_id | bill_date  | amount | paid
+---------+------------+--------+------
+ b1      | 2019-03-05 | 104238 | t
+ b2      | 2019-03-04 | 112632 | t
+ b3      | 2019-03-06 | 161496 | t
+ b4      | 2019-03-07 |  57539 | t
+ b5      | 2019-03-07 | 159787 | t
+ b6      | 2019-03-08 |  40635 | t
+ b7      | 2019-03-12 | 143923 | t
+ b8      | 2019-03-13 | 177501 | t
+ b9      | 2019-03-15 |  78625 | t
+ b10     | 2019-03-17 | 115190 | t
+ b11     | 2019-03-18 |  85369 | f
+ b12     | 2019-03-20 | 120970 | t
+ b13     | 2019-03-22 | 116608 | t
+ b14     | 2019-03-25 | 151407 | t
+ b15     | 2019-03-28 | 108224 | f
+ b16     | 2019-04-04 | 152525 | t
+ b17     | 2019-04-05 | 129935 | t
+ b18     | 2019-04-06 |  43422 | t
+ b19     | 2019-04-10 | 127151 | t
+ b20     | 2019-04-11 | 144057 | f
+ b21     | 2019-04-11 | 179716 | t
+ b22     | 2019-04-14 |  93821 | t
+ b23     | 2019-04-15 |  50077 | t
+ b24     | 2019-04-15 | 146789 | t
+ b25     | 2019-04-17 |  68022 | t
+ b26     | 2019-04-19 |  54575 | t
+ b27     | 2019-04-30 | 162245 | t
+(27 rows)
+
+
+1. How many patients visited the hospital in March and April respectively?
+
+postgres=# ;
+postgres=# SELECT COUNT(admit_date) from patient
+postgres-# WHERE admit_date >= '2019-03-01'::DATE
+postgres-# AND admit_date <= '2019-04-30'::DATE;
+ERROR:  relation "patient" does not exist
+LINE 1: SELECT COUNT(admit_date) from patient
+                                      ^
+postgres=# SELECT COUNT(admit_date) from patients
+postgres-# WHERE admit_date >= '2019-03-01'::DATE
+postgres-# AND admit_date <= '2019-04-30'::DATE;
+ count
+-------
+    27
+(1 row)
+
+
+2. What is the income of each doctor in the month of March and April respectively?:
+
+postgres=# SELECT patients.doctor_id, SUM(bills.amount)
+postgres-# FROM patients
+postgres-# JOIN bills ON patients.bill_id = bills.bill_id
+postgres-# WHERE bills.bill_date >= '2019-03-01'::DATE AND bills.bill_date <= '2019-04-30'::DATE
+postgres-# AND bills.paid = 't'
+postgres-# GROUP BY doctor_id;
+ doctor_id |     sum
+-----------+--------------
+ d1        | 1.054175e+06
+ d2        |       996182
+ d3        |       619847
+ d4        |        78625
+(4 rows)
+
+
+3. What is the total amount of bills that are pending for payment :
+
+postgres=# SELECT SUM(bills.amount)
+postgres-# FROM patients
+postgres-# JOIN bills ON patients.bill_id = bills.bill_id
+postgres-# WHERE bills.paid = 'f';
+  sum
+--------
+ 337650
+(1 row)
+
+
+4. What is the revenue generated by each department of the hospital in the month of March and April respectively?:
+
+postgres=# SELECT doctor.department, SUM(bills.amount)
+postgres-# FROM patients
+postgres-# JOIN bills ON patients.bill_id = bills.bill_id
+postgres-# JOIN doctor ON patients.doctor_id = doctor.doctor_id
+postgres-# WHERE bills.bill_date >= '2019-03-01'::DATE AND bills.bill_date <= '2019-04-30'::DATE
+postgres-# AND bills.paid = 't'
+postgres-# GROUP BY doctor.department;
+ department |     sum
```

Comment:
Can you split the revenue on a monthly and department basis. For example
Endo March Rs. x
Endo April Rs. y
Heart March Rs. A
Heart April Rs.B

Reply:
March:
department |  sum
------------+--------
 Endo       | 159787
 Heart      | 709832
 Ortho      | 592307
 Pneumonia  |  78625

April:
department |  sum
------------+--------
 Endo       | 460060
 Heart      | 344343
 Ortho      | 403875